iml-sandbox/Vagrantfile

```
@@ -515,13 +516,28 @@ def get_oss_block_device(slice, idxOrEmpty)
   SHELL
 end
 
+def get_vm_name(id)
+  out, err = Open3.capture2e("VBoxManage list vms")
+  raise "Unable to list vms" unless err.exitstatus === 0
+  
+  path = path = File.dirname(__FILE__).split('/').last
+  name = out.split(/\n/)
+     .select { |x| x.start_with? "\"#{path}_#{id}" }
+     .map { |x| x.tr('"', '')}
+     .map { |x| x.split(' ')[0].strip }
+     .first
+
+  name
+end
+
 # Checks if a scsi controller exists.
 # This is used as a predicate to create controllers, as vagrant does not provide this
 # functionality by default.
 def controller_exists(name, controller_name)
-  out, err = Open3.capture2e("VBoxManage showvminfo #{name}")
+  return false if name.nil?
 
-  return false if err.exitstatus != 0
+  out, err = Open3.capture2e("VBoxManage showvminfo #{name}")
+  raise "Unable to show vm info for #{name}" unless err.exitstatus === 0
```

Can we raise the original error here?

---

yeah